sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -410,15 +437,29 @@ void gnrc_lorawan_mcps_request(gnrc_lorawan_t *mac,
     int waiting_for_ack = mcps_request->type == MCPS_CONFIRMED;
 
     gnrc_lorawan_build_uplink(mac, pkt, waiting_for_ack,
-                              mcps_request->data.port);
+                              mcps_request->data.port, mcps_request->data.adr);
+
+    if (mac->mlme.pending_mlme_opts & GNRC_LORAWAN_MLME_OPTS_LINK_ADR_ANS) {
+        mac->mlme.pending_mlme_opts &= ~GNRC_LORAWAN_MLME_OPTS_LINK_ADR_ANS;
+        mac->mcps.adr_ack_cnt = 1;
+    }
 
     mac->mcps.waiting_for_ack = waiting_for_ack;
+
     mac->mcps.ack_requested = false;
 
     mac->mcps.nb_trials = CONFIG_LORAMAC_DEFAULT_RETX;
 
     mac->mcps.msdu = pkt;
-    mac->last_dr = mcps_request->data.dr;
+
+    DEBUG("LAST DR%d \n",mac->last_dr); //to be removed
+
+    if(!mcps_request->data.adr){
```

Comment:
as described above, this shouldn't be an mcps_request param. The upper layer should be able to turn ADR on or off using the corresponding API (mlme_set).